### PR TITLE
Add gcloud task version 0.3.

### PR DESCRIPTION
**Pull request by bendory**
# Changes

`gcloud` task version 0.3 always uses the latest `gcloud` CLI.

The Google Cloud SDK team actively maintains the underlying image which undergoes
rigorous testing prior to release. Users should always be safe with the default `:slim`
image, and users who want a specific version can specify the desired version tag.

For details on the underlying image, see https://github.com/GoogleCloudPlatform/cloud-sdk-docker

Changes from version 0.2:

- defaults to latest `:slim` image
- version and image are configurable via params
- added `SCRIPT` param to enable call of `gsutil` and other commands

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/gcloud/0.3/gcloud.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: gcloud
+  labels:
```
Comment (vinamra28) [inline task/gcloud/0.3/gcloud.yaml:5]: since we are running everything gcp, is there a way that we can test this Task?
/cc @afrittoli @bendory @jerop

**Review by vinamra28 — COMMENTED** [review]
/lgtm

File: task/gcloud/0.3/gcloud.yaml (hunk 1 above)
Comment (bendory) [reply task/gcloud/0.3/gcloud.yaml:5]: It certainly is testable -- at a minimum, `gcloud version` should succeed; `gcloud config list` could be compared to known golden output. I just looked in the `test` directory but I don't see any documentation on adding tests; is there a `README` or perhaps a sample test I can use as a guide?

Comment (vinamra28) [reply task/gcloud/0.3/gcloud.yaml:5]: probably you can refer the following https://github.com/tektoncd/catalog/tree/main/task/black/0.2 😅. Do let me know if you need any assistance, happy to help 🙂

Comment (bendory) [reply task/gcloud/0.3/gcloud.yaml:5]: Based on what I see in the `black` tests that you reference, I added a few tests -- but these are more like samples than tests, is there testing infrastructure for catalog or have I done what you're asking for?

I see other tasks have a `samples` directory, perhaps these belong in `samples`?

Comment (bendory) [reply task/gcloud/0.3/gcloud.yaml:5]: Note: the `config.yaml` and `version.yaml` I added are platform and environment agnostic. I could add samples or tests of GCP interactions, but they will be platform and environment dependent.

Comment (bendory) [reply task/gcloud/0.3/gcloud.yaml:5]: Aha! I just found the [guidance on e2e testing](https://github.com/tektoncd/catalog/blob/main/CONTRIBUTING.md#end-to-end-testing). Reverting to draft while I digest + implement that.

File: task/gcloud/0.3/README.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+  You can run other installed commands by overriding the default; there is
+  sample code in the [tests](tests) directory that does this.
+
+## Platforms
+
+The Task can be run on `linux/amd64` platform.
+
+## Usage
+
+### Authorizing `gcloud` commands
```
Comment (dibyom) [inline task/gcloud/0.3/README.md:47]: Thank you for the super detailed docs! This is one area that trips up a lot of users!

**Review by dibyom — APPROVED** [review]
Thank you!

/lgtm

**Review by vinamra28 — APPROVED** [review]
thanks for the PR 😄

**Pull request merged**